lib/p5.play.js

```diff
@@ -2291,32 +2334,20 @@ function Sprite(pInst, _x, _y, _w, _h) {
 
   this._findDisplacement = function(target) {
     // Multisample if tunneling occurs:
-    // If the direction between the two
-    // If the relative velocities of the two sprites is greater than the
-    // difference of their previous centers on the velocity axis,
-    // their centers might have crossed on an axis this frame which would permit
-    // tunneling.
+    // Do broad-phase detection. Check if the swept colliders overlap.
     // In that case, test interpolations between their last positions and their
     // current positions, and check for tunneling that way.
-    var relativeVelocity = p5.Vector.sub(this.velocity, target.velocity);
-    var squareVelocity = relativeVelocity.magSq();
-    var centerSquareDistanceOnVelocityAxis = p5.Vector.project(
-      p5.Vector.sub(this.collider.center, target.collider.center),
-      relativeVelocity
-    ).magSq();
-
-    // If the collider centers are closer on the velocity axis than the
-    // velocity itself, we are close enough to have tunneled this frame.
     // Use multisampling to catch collisions we might otherwise miss.
-    if (squareVelocity > centerSquareDistanceOnVelocityAxis) {
+    if (this._doSweptCollidersOverlap(target)) {
       // Figure out how many samples we should take.
       // We want to limit this so that we don't take an absurd number of samples
       // when objects end up at very high velocities (as happens sometimes in
       // game engines).
       var radiusOnVelocityAxis = Math.max(
         this.collider._getMinRadius(),
         target.collider._getMinRadius());
-      var timestep = Math.max(0.015, radiusOnVelocityAxis / Math.sqrt(squareVelocity));
+      var relativeVelocity = p5.Vector.sub(this.velocity, target.velocity).mag();
+      var timestep = Math.max(0.015, radiusOnVelocityAxis / relativeVelocity);
```

Comment:
Nit: I'd call it relativeSpeed rather than relativeVelocity, as it's a float quantity, not a vector